### ISSUE TITLE
issues with osism-update-manager

### ISSUE THREAD

**Issue by schulze-b1** [osism/issues#505]
while updating one manager node from osism 4.0 to 4.2, we encountered multiple issues:

- script seems to be in general not backwards compatible
- ansible version used in script is not possible to install due to ansible >= 7.0 needing python >= 3.9, which is not per default installed on ubuntu 20.04 (default is python 3.8) - so the virtalenv part in this script fails as well
- the ansible-galaxy parts also only work with ansible >= 7.0 (lower ansible versions complain about the git+https:// part)
- the osism ansible collections are always installed as latest. I could not see any tags corresponding to osism versions in the repo's. so is it safe to always use the latest collections even when using this with older osism versions?
- hosts-file/group_vars/host_vars-files in /opt/configuration/environments/manager/ are ignored and /opt/configuration/inventory is used instead
- the script also reads the files in /opt/configuration/environments/ - as you said in another issue, when upgrading the manager, the script should only use the configs under /opt/configuration/environments/manager/ and not both locations
- while manually fixing all these issues, there was also an pip error that wheel==0.40.0 was missing in requirements.txt

i was able to update the manager node only after installing python3.9 and using a virtualenv for python3.9 and temporarily fixing all other mentioned issues.

**Comment by berendt** [comment]
You probably have to set the Ansible version when working with Ubuntu 20.04. Via environment variables you can also set e.g. the versions of the Ansible collections. 

https://github.com/osism/ansible-collection-services/blob/main/roles/manager/templates/wrapper/osism-update-manager.j2#L5-L13

Also added some more parameters with https://github.com/osism/ansible-collection-services/pull/1033. This way you can also overwrite the used inventory/private key. Will be merged until the end of this week.

**Comment by berendt** [comment]
All points from the issue should be addressed by now.
